### PR TITLE
Update configuration.asciidoc

### PR DESCRIPTION
**Pull request by leahleahy**

### CODE REVIEW

File: docs/configuration.asciidoc [hunk 1]
```diff
@@ -24,7 +24,7 @@ When using this method, strings are split by comma, e.g.,
 [[configuration-precedence]]
 === Configuration precedence
 
-Options are applied in the following order (last one wins):
+Options are applied in the following order (last one wins):
```
Comment (leahleahy) [inline docs/configuration.asciidoc:27]: Should this say listed in descending order with the last taking precedence for consistency with other articles?

File: docs/configuration.asciidoc [hunk 2]
```diff
@@ -184,12 +184,11 @@ WARNING: Api keys only provide any real security if your APM server uses TLS.
...
-Maximum amount of objects kept in queue, before sending to APM Server.
+The maximum amount of objects kept in queue before sending to APM Server.
 
-If you hit this limit you either have to increase the agent's
-<<config-pool-size,worker pool size>> or it could mean the agent has trouble
-connecting to APM Server. The <<config-log-path,logs>> should tell you what
-went wrong.
+If the limit is hit, increase the agent's
+<<config-pool-size,worker pool size>> or the agent is having trouble
+connecting to APM Server. The <<config-log-path,logs>> should help to troubleshoot.
```
Comment (leahleahy) [inline docs/configuration.asciidoc:191]: Is this better than saying "should tell you what went wrong" ?

File: docs/configuration.asciidoc [hunk 3]
```diff
@@ -508,7 +504,7 @@ Use this option to ignore certain URL patterns eg. healthchecks or admin section
 | `ELASTIC_APM_INSTRUMENTED_RAKE_TASKS` | `instrumented_rake_tasks` | `[]`    | `['my_task']`
 |============
 
-Elastic APM can instrument your Rake tasks but given that they are used for a multitude of sometimes very different and not always relevant things, this is opt in.
+Elastic APM can instrument your Rake tasks. Theis is an opt-in field, as they are used are for a multitude of things.
```
Comment (leahleahy) [inline docs/configuration.asciidoc:507]: Does this make sense or is it changing the meaning too much?

File: docs/configuration.asciidoc [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
-https://www.elastic.co/guide/en/apm/agent/ruby/current/introduction.html[elastic.co]
+https://www.elastic.co/guide/en/apm/agent/ruby/current/introduction.html[elastic.co].
 endif::[]
 
 [[configuration]]
 == Configuration
 
 There are several ways to configure how Elastic APM behaves.
-The recommended way is to specify options in a `config/elastic_apm.yml` file:
+We recommend the method of specifying options in a `config/elastic_apm.yml` file:
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:11]:
```suggestion
We recommend using a `config/elastic_apm.yml` file:
```

File: docs/configuration.asciidoc [hunk 5]
```diff
@@ -37,7 +37,7 @@ Options are applied in the following order (last one wins):
 [[dynamic-configuration]]
 === Dynamic configuration
 
-Configuration options marked with the image:./images/dynamic-config.svg[] badge can be changed at runtime
+Configuration options that are marked with the image:./images/dynamic-config.svg[] badge can be changed at runtime
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:40]:
```suggestion
Configuration options marked with the image:./images/dynamic-config.svg[] badge can be changed at runtime
```

File: docs/configuration.asciidoc [hunk 6]
```diff
@@ -59,7 +59,7 @@ config.elastic_apm.service_name = 'MyApp'
 [float]
 === Sinatra and Rack
 
-When using Sinatra and Rack, you can configure when starting
+When using Sinatra and Rack, configure when starting
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:62]:
```suggestion
When using Sinatra and Rack, you can configure when starting
```

File: docs/configuration.asciidoc [hunk 7]
```diff
@@ -150,7 +150,7 @@ and port.
 
 This string is used to ensure that only your agents can send data to your APM server.
 Both the agents and the APM server have to be configured with the same secret token.
-One example to generate a secure secret token is:
+An example to generate a secure secret token is:
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:153]:
```suggestion
Here's an example that generates a secure secret token:
```

File: docs/configuration.asciidoc [hunk 8]
```diff
@@ -170,11 +170,11 @@ WARNING: Secret tokens only provide any real security if your APM server uses TL
 |============
 
 This base64-encoded string is used to ensure that only your agents can send data to your APM server.
-You must have created the api key using the APM server command line tool. Please see the APM server
-documentation for details on how to do that. Please note that this feature is experimental in the
+The API key must be created using the APM server command line tool. Please see the APM server
+documentation for details on how to create the API key. Please note that this feature is experimental in the
 APM server in version 7.6.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:175]:
```suggestion
The API key must be created using the {apm-server-ref-v}/api-key.html[APM server command-line tool]. 
```

File: docs/configuration.asciidoc (hunk 2 above)
Comment (bmorelli25) [inline docs/configuration.asciidoc:191]:
```suggestion
If you hit the limit, consider increasing the agent's <<config-pool-size,worker pool size>>.
If you don't, the agent may have trouble connecting to APM Server.
The <<config-log-path,logs>> should tell you what went wrong.
```

File: docs/configuration.asciidoc [hunk 9]
```diff
@@ -233,8 +230,8 @@ It has to be provided in *<<config-format-duration, duration format>>*.
 | `ELASTIC_APM_BREAKDOWN_METRICS` | `breakdown_metrics` | `true`
 |============
 
-Enable/disable the tracking and collection of breakdown metrics.
-By setting this to `False`, tracking this metric is completely disabled, which can reduce the overhead of the agent.
+Enable or disable the tracking and collection of breakdown metrics.
+By setting this to `False`, the tracking is disabled, which can reduce the overhead of the agent.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:234]:
```suggestion
Setting this to `False` disables the tracking of breakdown metrics, which can reduce the overhead of the agent.
```

File: docs/configuration.asciidoc [hunk 10]
```diff
@@ -272,7 +268,7 @@ If your service handles data like this, we advise to only enable this feature wi
 | `ELASTIC_APM_CAPTURE_HEADERS` | `capture_headers` | `true`
 |============
 
-Whether or not to attach the request headers to transactions and errors.
+This indicates whether or not to attach the request headers to transactions and errors.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:271]:
```suggestion
Configure whether or not to attach the request headers to transactions and errors.
```

File: docs/configuration.asciidoc [hunk 11]
```diff
@@ -302,10 +298,10 @@ Whether or not to attach `ENV` from Rack to transactions and errors.
 | `ELASTIC_APM_CENTRAL_CONFIG` | `central_config` | `true`
 |============
 
-Enables {apm-app-ref}/agent-configuration.html[APM Agent Configuration via Kibana].
+This enables {apm-app-ref}/agent-configuration.html[APM Agent Configuration via Kibana].
 If set to `true`, the client will poll the APM Server regularly for new agent configuration.
 
-Usually APM Server determines how often to poll, but if not the default interval is 5 minutes.
+Usually APM Server determines how often to poll, but if not set the default interval is 5 minutes.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:304]:
```suggestion
Usually APM Server determines how often to poll, but if not, set the default interval is 5 minutes.
```

File: docs/configuration.asciidoc (hunk 3 above)
Comment (bmorelli25) [reply docs/configuration.asciidoc:507]:
```suggestion
Elastic APM can instrument your Rake tasks. Because Rake tasks are used for very different things, this is an opt-in field.
```

File: docs/configuration.asciidoc [hunk 12]
```diff
@@ -639,9 +635,9 @@ when this option is false.
 | `ELASTIC_APM_SANITIZE_FIELD_NAMES` | `sanitize_field_names` | `"password,passwd,pwd,secret,*key,*token*,*session*,*credit*,*card*,authorization,set-cookie"` | `Auth*tion,abc*,*xyz`
 |============
 
-Sometimes it is necessary to sanitize the data sent to Elastic APM, e.g. remove sensitive data.
+Sometimes it is necessary to sanitize the data sent to Elastic APM to remove sensitive data.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:638]:
```suggestion
Sometimes it is necessary to sanitize the data sent to Elastic APM to remove sensitive values.
```

File: docs/configuration.asciidoc [hunk 13]
```diff
@@ -657,16 +653,15 @@ Matching is case insensitive.
 | `ELASTIC_APM_SERVICE_NAME` | `service_name` | App's name | `MyApp`
 |============
 
-The name of your service.
-This is used to keep all the errors and transactions of your service together and is
+The name of your service. This is used to group the errors and transactions of your service and is the
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:656]:
```suggestion
The name of your service. This is used to group the errors and transactions of your service and is
```

File: docs/configuration.asciidoc [hunk 14]
```diff
@@ -657,16 +653,15 @@ Matching is case insensitive.
...
+The name of your service. This is used to group the errors and transactions of your service and is the 
 the primary filter in the Elastic APM user interface.
 
 If you're using Ruby on Rails this will default to your app's name.
 If you're using Sinatra it will default to the name of your app's class.
 
 NOTE: The service name must conform to this regular expression: `^[a-zA-Z0-9 _-]+$`.
-In layman's terms: Your service name must only contain characters from the ASCII
-alphabet, numbers, dashes, underscores and spaces.
+Your service name must only contain characters from the ASCII
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:663]:
```suggestion
In other words, it must only contain characters from the ASCII
```

File: docs/configuration.asciidoc [hunk 15]
```diff
@@ -857,14 +851,14 @@ The unit is provided as suffix directly after the number, without any separation
...
 
-NOTE: we use the power-of-two sizing convention, e.g. `1 kilobyte == 1024 bytes`
+NOTE: We use the power-of-two sizing convention, e.g. `1 kilobyte == 1024 bytes`.
 
 [float]
 [[special-configuration]]
 === Special configuration
 
-Elastic APM patches `Kernel#require` to auto-detect and instrument supported third party libraries. It does so with the utmost care but in rare cases it can conflict with some libraries.
+Elastic APM patches `Kernel#require` to auto-detect and instrument supported third party libraries. However, it may conflict with some libraries.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:860]:
```suggestion
Elastic APM patches `Kernel#require` to auto-detect and instrument supported third party libraries. It does so with the utmost care but in rare cases it can conflict with some libraries.
```

**Comment by bmorelli25** [comment]
Jenkins, run the tests please

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Review by bmorelli25 — APPROVED** [review]
Pending doc build, LGTM

**Review by mikker — APPROVED** [review]
These are good suggestions, thanks!
